shared/js/smart-screen/f_t_e_wizard.js

```diff
@@ -0,0 +1,192 @@
+    }
+  });
+
+  Object.defineProperty(FTEWizard.prototype, 'finished', {
+    get: function() {
+      return this._finished;
+    }
+  });
+
+  FTEWizard.prototype.init = function fw_init(options) {
+    if (this._launched) {
+      if (options.container) {
+        options.container.style.display = 'none';
+      }
+      options.onfinish && options.onfinish();
+      return;
+    }
+
+    this._container = options.container;
+    this._buttonsClass = options.buttonsClass || 'fte-button';
+    this._finishButtonClass = options.finishButtonClass || 'fte-finish';
+    this._pageClass = options.pageClass || 'fte-page';
+    this.propagateKeyEvent = options.propagateKeyEvent || false;
+    this._simpleKeyNavigation = new SimpleKeyNavigation();
+    this._onfinish = options.onfinish;
+
+    this._pages =
+            Array.from(this._container.getElementsByClassName(this._pageClass));
+    this._nextButton =
+               this._container.getElementsByClassName(this._nextButtonClass)[0];
+    this._prevButton =
+               this._container.getElementsByClassName(this._prevButtonClass)[0];
+    this._finishButton =
+             this._container.getElementsByClassName(this._finishButtonClass)[0];
+    this._currentPage = 0;
+
+
+    this._container.addEventListener('keydown', this);
+    this._container.addEventListener('keyup', this);
+
+    // The element list will be updated in this._updateNavigation.
+    this._simpleKeyNavigation.start([],
+                                     SimpleKeyNavigation.DIRECTION.HORIZONTAL);
+    this._pages.forEach((page, idx) => this._hide(idx));
+    this._show(this._currentPage);
+    this._updateNavigation();
+
+    this._running = true;
+  };
+
+  FTEWizard.prototype.uninit = function fw_uninit() {
+    if (this._launched) {
+      return;
+    }
+    this._container.removeEventListener('keydown', this);
+    this._container.removeEventListener('keyup', this);
+    this._simpleKeyNavigation.stop();
+
+    this._running = false;
+  };
+
+  FTEWizard.prototype.goNext = function fw_next() {
+    if (this._currentPage >= this._pages.length - 1) {
+      this.finish();
+      return;
+    }
+
+    this._hide(this._currentPage);
+    this._currentPage++;
+    this._show(this._currentPage);
+    this._updateNavigation();
+  };
+
+  FTEWizard.prototype.goPrev = function fw_prev() {
+    if (this._currentPage <= 0) {
+      return;
+    }
+
+    this._hide(this._currentPage);
+    this._currentPage++;
+    this._show(this._currentPage);
+    this._updateNavigation();
+  };
+
+  FTEWizard.prototype.finish = function fw_finish() {
+    this._container.classList.add('finish');
+
+    attachTransitionEnd(this._container, () => {
+
+      this._hide(this._currentPage);
+      this._container.style.display = 'none';
+      this.uninit();
+      this._finished = true;
+      this._launched = true;
+
+      localStorage.setItem(this._name + '.fteskip', true);
+      this._onfinish && this._onfinish();
```

Comment:
I guess `this._onfinish` becomes useless after this line. How about adding `this._onfinish = null` here?